Compilation problem in virtualenv "cannot find -lInclude"

Lacking root permissions on Ubuntu 12.04 I compiled the recent Python 3.5 from source, setup a virtualenv and pip-ed Theano. 
When running ``import theano`` I get an error: ``/usr/bin/ld: cannot find -lInclude`. The flag is attached to the following compile statement:

```
Problem occurred during compilation with the command line below:
/usr/bin/g++ -shared -g -march=corei7-avx -mcx16 -msahf -mno-movbe -maes -mpclmul -mpopcnt -mno-abm -mno-lwp -mno-fma -mno-fma4 -mno-xop -mno-bmi -mno-tbm -mavx -msse4.2 -msse4.1 -mno-rdrnd -mno-f16c -mno-fsgsbase --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=20480 -mtune=generic -D NPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -m64 -fPIC -I/path/to/virtualenv/venv/lib/python3.5/site-packages/numpy/core/include -I/path/to/virtualenv/Python-3.5.0/Include -I/path/to/virtualenv/Python-3.5.0 -o /home/fgressmann/.theano/compiledir_Linux-3.2--generic-x86_64-with-debian-wheezy-sid-x86_64-3.5.0-64/lazylinker_ext/lazylinker_ext.so /home/fgressmann/.theano/compiledir_Linux-3.2--generic-x86_64-with-debian-wheezy-sid-x86_64-3.5.0-64/lazylinker_ext/mod.cpp -L/usr/local/lib -lInclude
```
Any ideas how to fix it?

The complete traceback:
```
ImportError                               Traceback (most recent call last)
/path/to/virtualenv/venv/lib/python3.5/site-packages/theano/gof/lazylinker_c.py in <module>()
     63         if version != getattr(lazylinker_ext, '_version', None):
---> 64             raise ImportError()
     65 except ImportError:

ImportError: 

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
/path/to/virtualenv/venv/lib/python3.5/site-packages/theano/gof/lazylinker_c.py in <module>()
     80             if version != getattr(lazylinker_ext, '_version', None):
---> 81                 raise ImportError()
     82         except ImportError:

ImportError: 

During handling of the above exception, another exception occurred:

Exception                                 Traceback (most recent call last)
<ipython-input-1-1a165bb0dd56> in <module>()
      1 
----> 2 import theano

/path/to/virtualenv/venv/lib/python3.5/site-packages/theano/__init__.py in <module>()
     53     object2, utils
     54 
---> 55 from theano.compile import \
     56     SymbolicInput, In, \
     57     SymbolicOutput, Out, \

/path/to/virtualenv/venv/lib/python3.5/site-packages/theano/compile/__init__.py in <module>()
      7         SpecifyShape, specify_shape, register_specify_shape_c_code)
      8 
----> 9 from theano.compile.function_module import *
     10 
     11 from theano.compile.mode import *

/path/to/virtualenv/venv/lib/python3.5/site-packages/theano/compile/function_module.py in <module>()
     16 from theano import gof
     17 from theano.compat.python2x import partial
---> 18 import theano.compile.mode
     19 from theano.compile.io import (
     20     In, SymbolicInput, SymbolicInputKit, SymbolicOutput)

/path/to/virtualenv/venv/lib/python3.5/site-packages/theano/compile/mode.py in <module>()
      9 import  theano
     10 from theano import gof
---> 11 import theano.gof.vm
     12 from theano.configparser import config, AddConfigVar, StrParam
     13 from theano.compile.ops import register_view_op_c_code, _output_guard

/path/to/virtualenv/venv/lib/python3.5/site-packages/theano/gof/vm.py in <module>()
    566 
    567 try:
--> 568     from . import lazylinker_c
    569 
    570     class CVM(lazylinker_c.CLazyLinker, VM):

/path/to/virtualenv/venv/lib/python3.5/site-packages/theano/gof/lazylinker_c.py in <module>()
    114             args = cmodule.GCC_compiler.compile_args()
    115             cmodule.GCC_compiler.compile_str(dirname, code, location=loc,
--> 116                                              preargs=args)
    117             # Save version into the __init__.py file.
    118             init_py = os.path.join(loc, '__init__.py')

/path/to/virtualenv/venv/lib/python3.5/site-packages/theano/gof/cmodule.py in compile_str(module_name, src_code, location, include_dirs, lib_dirs, libs, preargs, py_module)
   2008             # difficult to read.
   2009             raise Exception('Compilation failed (return status=%s): %s' %
-> 2010                             (status, compile_stderr.replace('\n', '. ')))
   2011         elif config.cmodule.compilation_warning and compile_stderr:
   2012             # Print errors just below the command line.

Exception: Compilation failed (return status=1): /usr/bin/ld: cannot find -lInclude. collect2: ld returned 1 exit status. 
```

Yes, the path is existing, but changing the line to ``libname = python_inc`` only brings up an analog error: ``/usr/bin/ld: cannot find -l/path/to/local/compiled/Python-3.5.0/Include``

--- a/theano/gof/cmodule.py
+++ b/theano/gof/cmodule.py
@@ -1,4 +1,4 @@
-"""
+rator"""
 Generate and compile C modules for Python.
 
 """
@@ -1649,9 +1649,9 @@ def std_lib_dirs_and_libs():
     elif sys.platform == 'darwin':
         std_lib_dirs_and_libs.data = [], []
     else:
-        # assume Linux 
+        # assume Linux
         # Typical include directory: /usr/include/python2.6
-        
+
         # get the name of the python library (shared object)
         libname = distutils.sysconfig.get_config_var("LDLIBRARY")
 
@@ -1661,10 +1661,10 @@ def std_lib_dirs_and_libs():
         # remove extension if present
         if libname.endswith(".so"):
             libname = libname[:-3]
-    
+
         libdir = distutils.sysconfig.get_config_var("LIBDIR")
 
-        std_lib_dirs_and_libs.data = [ libname ], [ libdir ]
+        std_lib_dirs_and_libs.data = [libname], [libdir]
 
     # sometimes, the linker cannot find -lpython so we need to tell it
     # explicitly where it is located this returns

--- a/theano/gof/cmodule.py
+++ b/theano/gof/cmodule.py
@@ -1649,9 +1649,22 @@ def std_lib_dirs_and_libs():
     elif sys.platform == 'darwin':
         std_lib_dirs_and_libs.data = [], []
     else:
+        # assume Linux 
         # Typical include directory: /usr/include/python2.6
-        libname = os.path.basename(python_inc)
-        std_lib_dirs_and_libs.data = [libname], []
+        
+        # get the name of the python library (shared object)
+        libname = distutils.sysconfig.get_config_var("LDLIBRARY")
+
+        if libname.startswith("lib"):
+            libname = libname[3:]
+
+        # remove extension if present
+        if libname.endswith(".so"):
+            libname = libname[:-3]
+    
+        libdir = distutils.sysconfig.get_config_var("LIBDIR")
+
+        std_lib_dirs_and_libs.data = [ libname ], [ libdir ]
 
     # sometimes, the linker cannot find -lpython so we need to tell it
     # explicitly where it is located this returns

--- a/theano/gof/cmodule.py
+++ b/theano/gof/cmodule.py
@@ -1,4 +1,4 @@
-rator"""
+"""
 Generate and compile C modules for Python.
 
 """

